src/pybind/mgr/dashboard/module.py

```diff
@@ -914,13 +954,24 @@ def perf(self, osd_id):
                     ceph_version=global_instance().version,
                     path_info='/osd' + cherrypy.request.path_info,
                     toplevel_data=json.dumps(toplevel_data, indent=2),
-                    content_data=json.dumps(self._osd(osd_id), indent=2)
+                    content_data=json.dumps(self.perf_data(osd_id), indent=2)
                 )
 
             @cherrypy.expose
             @cherrypy.tools.json_out()
             def perf_data(self, osd_id):
-                return self._osd(osd_id)
+                try:
+                    osd_id = int(osd_id)
+                except ValueError:
+                    raise cherrypy.HTTPError(400,
+                        "Invalid OSD id {0}".format(osd_id))
+
+                r = self._osd(osd_id)
+                if r is None:
```

Comment:
it's okay to push the HTTPError etc exceptions down into functions like `_osd` -- it's more pythonic than returning Nones to indicate something missing